src/main/groovy/edu/oregonstate/mist/recommendations/db/RecommendationDAO.groovy

```diff
@@ -0,0 +1,117 @@
+package edu.oregonstate.mist.recommendations.db
+
+import edu.oregonstate.mist.recommendations.core.Recommendation
+import edu.oregonstate.mist.recommendations.mapper.RecommendationMapper
+import org.skife.jdbi.v2.sqlobject.Bind
+import org.skife.jdbi.v2.sqlobject.SqlQuery
+import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper
+
+/**
+ * Recommendation DAO
+ */
+@RegisterMapper(RecommendationMapper)
+public interface RecommendationDAO extends Closeable {
+
+    @SqlQuery("""
+              SELECT * FROM (
+                SELECT T.*, ROWNUM AS R_NUM
+                FROM (
+                    SELECT UNIVERSITY.NAME AS UNIVERSITY,
+                           UNIVERSITY.IS_985,
+                           UNIVERSITY.IS_211,
+                           ENROLLMENT_UNIVERSITY.YEAR,
+                           ENROLLMENT_UNIVERSITY.SCORE_AVG,
+                           ENROLLMENT_UNIVERSITY.STU_COUNT,
+                           RANKING.RANK,
+                           BATCH_SCORE.MIN_SCORE AS BATCH_SCORE,
+                           (ENROLLMENT_UNIVERSITY.SCORE_AVG - BATCH_SCORE.MIN_SCORE) AS SCORE_DIFF,
+                           NULL AS MAJOR
+                    FROM ENROLLMENT_UNIVERSITY
+                      LEFT JOIN RANKING
+                      ON RANKING.YEAR = ENROLLMENT_UNIVERSITY.YEAR AND
+                         RANKING.SCORE = ENROLLMENT_UNIVERSITY.SCORE_AVG AND
+                         RANKING.STUDENT_POOL_ID = ENROLLMENT_UNIVERSITY.STUDENT_POOL_ID
+                      LEFT JOIN BATCH_SCORE
+                      ON BATCH_SCORE.STUDENT_POOL_ID = ENROLLMENT_UNIVERSITY.STUDENT_POOL_ID AND
+                         BATCH_SCORE.YEAR = ENROLLMENT_UNIVERSITY.YEAR
+                      LEFT JOIN UNIVERSITY
+                      ON UNIVERSITY.ID = ENROLLMENT_UNIVERSITY.UNIVERSITY_ID
+                    WHERE
+                       ENROLLMENT_UNIVERSITY.STUDENT_POOL_ID = (
+                          SELECT STUDENT_POOL.ID FROM STUDENT_POOL
+                          WHERE PROVINCE = :province AND
+                                STU_TYPE = :studentType AND
+                                BATCH = :batch
+                         ) AND
+                         RANK >= :minRank AND
+                         RANK <= :maxRank AND
+                         ENROLLMENT_UNIVERSITY.YEAR like '%' ||:year|| '%'
```

Comment:
year is a optional parameter, and  when year is not provided, it should return records of all years.